packages/calypso-products/src/plans-utilities.ts

```diff
@@ -41,10 +41,40 @@ export function getTermDuration( term: string ): number | undefined {
 
 export const redirectCheckoutToWpAdmin = (): boolean => !! JETPACK_REDIRECT_CHECKOUT_TO_WPADMIN;
 
-export const is2023PricingGridEnabled = () => {
-	const supportedLocales = [ 'en', 'en-gb', 'es' ];
-	return (
-		isEnabled( 'onboarding/2023-pricing-grid' ) &&
-		supportedLocales.includes( getLocaleSlug() ?? '' )
-	);
+/**
+ * Returns if the 2023 Pricing grid optimized design should be visible or not
+ *
+ */
+export const is2023PricingGridEnabled = (): boolean => {
+	const isFeatureFlagEnabled = (): boolean => isEnabled( 'onboarding/2023-pricing-grid' );
+
+	const isLocaleSupported = (): boolean => {
+		const supportedLocales = [ 'en', 'en-gb', 'es' ];
+		return supportedLocales.includes( getLocaleSlug() ?? '' );
+	};
+
+	const isActivePage = (): boolean => {
```

Comment:
Probably best to move this function definition to the outer/module scope. No need to be redefining these every time `is2023PricingGridEnabled` gets called. Same for `isLocaleSupported`.

Reply:
I have done this experiment before and moving these definitions inside the function seem to not have much of an impact. for 100_000 clicks the difference is in milliseconds. And most of the time the experiment has favourable time for having the functions inside. But I am not sure we can derive any meaningful performance stats at this minute difference level. At the very least it might not be that impactful. 

https://jsfiddle.net/jdc91xt/cu8s7gdk/40/

<img width="1722" alt="image" src="https://user-images.githubusercontent.com/3422709/220011694-864fa47d-d805-49e8-bd49-5c07cced4742.png">


Consequently, I think it's better to have the sub functions inside this function for better readability. Do let me know if there is a flaw in the way I am measuring these information 👍

Reply:
Not sure I find that more readable really, but as you like.

I think it's a question of good practice, drawing a line irrespective of minuscule or bigger gains. Unless memory is a concern, then the engine probably has to do additional work. It might also be a way to keep the code modular and give it proper names. For one, `isActivePage` is ambiguous, whether it lives here or outside (if outside, you'd probably be giving it a better name to account for the separation also - something like `isPathEnabledFor2023Grid` perhaps).

no strong feelings. I also do what you did many times, just probably should stop.